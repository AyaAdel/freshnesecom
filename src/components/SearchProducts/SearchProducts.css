.search-products {
  display: flex;
  align-items: center;
  background: #f9f9f9;
  border: 1px solid #d1d1d1;
  border-radius: 12px;
  width: 500px;
  height: 42px;
  padding: 10px 16px;
}

.search-products__input {
  flex-grow: 1;
  border: none;
  font-size: 14px;
  line-height: 19px;
  margin: 0 24px;
  background-color: transparent;
}

.search-products__input:focus {
  outline: 0;
}

.search-products .dropdown-list__list__selected {
  font-weight: 700;
  font-size: 15px;
  line-height: 22px;
  color: #151515;
  position: relative;
  padding-right: 24px;
  border-right: 1px solid #d1d1d1;
  width: 160px;
}

.dropdown-list__list__selected span {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.search-products__icon {
  width: 16px;
  height: 16px;
}

.header__right-side {
  display: flex;
  align-items: center;
  gap: 43px;
}

.header__right-side .icon {
  width: 24px;
  height: 24px;
}

.search-products .dropdown-list__items {
  left: -16px;
}

.search-products .dropdown-list__list__selected img {
  width: 16px;
}
