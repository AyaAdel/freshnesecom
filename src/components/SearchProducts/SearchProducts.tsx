import { useState } from 'react';
import SearchIcon from 'assets/search.svg';
import { searchProducts, getProductsOfCategory } from 'services/products';
import Dropdown from 'components/Dropdown/Dropdown';
import { useAppSelector, useAppDispatch } from 'app/hooks';
import type { RootState, AppDispatch } from 'app/store';
import { loadProducts } from 'features/products/productsSlice';

import './SearchProducts.css';

export default function SearchProducts() {
  const [value, setValue] = useState('');
  const categories: string[] = useAppSelector(
    (state: RootState) => state.categories
  );

  const dispatch: AppDispatch = useAppDispatch();

  const filterProducts = async (event: React.ChangeEvent<HTMLInputElement>) => {
    const productName = event.target.value;

    setValue(productName);
    const response = await searchProducts(productName);
    dispatch(loadProducts(response));
  };

  const filterProductsWithCategory = async (category: string) => {
    const response = await getProductsOfCategory(category);
    dispatch(loadProducts(response));
  };

  return (
    <div className="search-products">
      <Dropdown
        defaultSelected="All Categories"
        list={categories}
        onSelect={filterProductsWithCategory}
      />
      <input
        className="search-products__input"
        placeholder="Search Products, categories ..."
        onChange={filterProducts}
        value={value}
      />
      <img src={SearchIcon} className="search-products__icon" />
    </div>
  );
}
