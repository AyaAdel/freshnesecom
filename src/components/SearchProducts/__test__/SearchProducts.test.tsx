import { render, screen, waitFor, act, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { RenderComponent } from 'utils/test/render';
import SearchProducts from '../SearchProducts';

function MyComponent() {
  return (
    <RenderComponent>
      <SearchProducts />
    </RenderComponent>
  );
}

it('Render Search Categories Correctly', () => {
  const tree = renderer.create(<MyComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
