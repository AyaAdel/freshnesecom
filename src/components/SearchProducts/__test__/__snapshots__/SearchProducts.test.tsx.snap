// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Search Categories Correctly 1`] = `
<div
  className="search-products"
>
  <div
    className="dropdown-list"
    data-testid="dropdown-list"
  >
    <div
      className="dropdown-list__list__selected"
      onClick={[Function]}
    >
      <span
        data-testid="dropdown-list-selected"
      >
        All Categories
      </span>
      <img
        src="chevron-down.svg"
      />
    </div>
  </div>
  <input
    className="search-products__input"
    onChange={[Function]}
    placeholder="Search Products, categories ..."
    value=""
  />
  <img
    className="search-products__icon"
    src="search.svg"
  />
</div>
`;
